File: java/src/main/java/publicdomain/test/snippets/system/Time.java
package publicdomain.test.snippets.system;

import java.time.Instant;

/**
 * {@link Time} provides an abstraction of the current date/time. The values provided by this interface may be affected
 * by changes in the system, such as user intervention or time-synchronization.
 */
public interface Time {
    /**
     * @return the current date/time.
     */
    Instant now();
}

File: dotnet/TestSnippets.ForProduction/System/Time.cs
using System;

namespace PublicDomain.TestSnippets.System
{
    /// <summary>
    /// <see cref="Time"/> provides an abstraction of the current date/time.
    /// The values provided by this interface may be affected by changes in the system,
    /// such as user intervention or time-synchronization.
    /// </summary>
    public interface Time
    {
        /// <summary>
        /// The current date/time in UTC.
        /// </summary>
        DateTime Now { get; }
    }
}